runtime-spec/parachain/messaging-spec.md

```diff
@@ -0,0 +1,380 @@
+implementing cross-chain interoperability. The upcoming replacement of HRMP is
+Cross-Chain Message Passing (XCMP), which exchanges messages directly between
+parachains and only updates proofs and read-confirmations on chain. With XCMP,
+vertical message processing is only used for opening and closing channels.
+
+### Channels
+
+A channel is a construct on the relay chain indicating an open, one-directional
+communication line between a sender and a recipient, including information about
+how the channel is being used. The channel itself does not contain any messages.
+A channel construct is created for each, individual communication line.
+
+A channel contains the following fields:
+
+`HrmpChannel`:
+
+- `sender_deposit: int`: staked balances of sender.
+- `recipient_deposit: int`: staked balances of recipient.
+- `limit_used_places: int`: the maximum number of messages that can be pending
+  in the channel at once.
+- `limit_used_bytes: int`: the maximum total size of the messages that can be
+  pending in the channel at once.
+- `limit_message_size`: the maximum message size that could be put into the
+  channel.
+- `used_places: int`: number of messages used by the sender in this
+  channel.
+- `used_bytes: int`: total number of bytes used by the sender in this
+  channel.
+- `sealed: bool`: (TOOD: this is not defined in the Impl-Guide) indicator wether
+  the channel is sealed. If it is, then the recipient will no longer
+  accept any new messages.
+- `mqc_head`: a head of the MQC for this channel.
+
+This structure is created or overwritten on every start of each session.
+Individual fields of this construct are updated for every message sent, such as
+`used_places`, `used_bytes` and `mqc_head`. If the channel is sealed and
+`used_places` reaches `0` (occurs when a new session begins), this construct is
+be removed on the *next* session start.
+
+The Runtime maintains a structure of the current, open channels in a map. The
+key is a tuple of the sender ParaId and the recipient ParaId, where the value is
+the corresponding `HrmpChannel` structure.
+
+```
+channels: map(ParaId, ParaId) => Channel
+```
+
+### Opening Channels
+
+Polkadot places a certain limit on the amount of channels that can be opened
+between parachains. Only the the sender can open a channel.
+
+In order to open a channel, the sender must send an opening request
+to the relay chain. The request is a construct containing the following fields:
+
+`ChOpenRequest`:
+
+- `sender: ParaId`: the ParaId of the sender.
+- `recipient: ParaId`: the ParaId of the recipient.
+- `confirmed: bool`: indicated whether the recipient has accepted the channel.
+  On request creation, this value is `false`.
+- `age: int`: the age of this request, which start at `0` and is incremented by
+  1 on every session start.
+
+TODO: Shouldn't `ChOpenRequest` also have an `initiator` field? Or can only the
+sender open an channel?
```

Comment:
Above you say, only the sender can open a channel, so we can probably drop this.